system/jlib/jtrace.cpp

```diff
@@ -107,6 +108,280 @@ class NoopSpanExporterFactory
+    }
+}
+
+/**
+ * Converts an OpenTelemetry span kind to its string representation.
+ * These are OTel span kinds defined in include/opentelemetry/trace/span_metadata.h,
+ * not HPCC JLib CSpan kinds
+ *
+ * @param spanKind The OpenTelemetry span kind to convert.
+ * @return The string representation of the OpenTelemetry span kind.
+ */
+static const char * spanKindToString(opentelemetry::trace::SpanKind spanKind)
+{
+    switch (spanKind)
+    {
+        case opentelemetry::trace::SpanKind::kClient:
+            return "Client";
+        case opentelemetry::trace::SpanKind::kServer:
+            return "Server";
+        case opentelemetry::trace::SpanKind::kProducer:
+            return "Producer";
+        case opentelemetry::trace::SpanKind::kConsumer:
+            return "Consumer";
+        case opentelemetry::trace::SpanKind::kInternal:
+            return "Internal";
+        default:
+            return "Unknown";
+    }
+}
+
+class JLogSpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+public:
+    JLogSpanExporter(SpanLogFlags spanLogFlags) : logFlags(spanLogFlags), shutDown(false) {}
+
+    /**
+    * @return Returns a unique pointer to an empty recordable object
+    */
+    virtual std::unique_ptr<opentelemetry::sdk::trace::Recordable> MakeRecordable() noexcept override
+    {
+        return std::unique_ptr<opentelemetry::sdk::trace::Recordable>(new opentelemetry::sdk::trace::SpanData());
+    }
+
+    /**
+    * Export - Formats recordable spans in HPCC Jlog format and reports to JLog
+    *
+    * @param recordables
+    * @return Always returns success
+    */
+    opentelemetry::sdk::common::ExportResult Export(
+      const nostd::span<std::unique_ptr<opentelemetry::sdk::trace::Recordable>> &recordables) noexcept override
+    {
+        if (isShutDown())
+        {
+            //Not logging this event per code review comments
+            return opentelemetry::sdk::common::ExportResult::kFailure;
+        }
+
+        for (auto &recordable : recordables)
+        {
+            auto span = std::unique_ptr<opentelemetry::sdk::trace::SpanData>(
+            static_cast<opentelemetry::sdk::trace::SpanData *>(recordable.release()));
+
+            if (span != nullptr)
+            {
+                char traceID[32]       = {0};
+                char spanID[16]        = {0};
+
+                span->GetTraceId().ToLowerBase16(traceID);
+                span->GetSpanId().ToLowerBase16(spanID);
+
+                StringBuffer out;
+                out.setf("{ \"name\": \"%s\"", span->GetName().data());
+                out.append(", \"trace_id\": \"").append(32, traceID).append("\"");
+                out.append(", \"span_id\": \"").append(16, spanID).append("\"");
+                out.appendf(", \"start\": %s", std::to_string(span->GetStartTime().time_since_epoch().count()).c_str());
+                out.appendf(", \"duration\": %s", std::to_string(span->GetDuration().count()).c_str());
```

Comment:
We talked about possibly needing to cast the duration, but that function returns chrono::nanos:

std::chrono::nanoseconds opentelemetry::v1::sdk::trace::SpanData::GetDuration